text/0000-Project-Goals-2024h2.md

```diff
@@ -0,0 +1,522 @@
+
+### crates-io team
+| Goal                                                                                  | Owner      | Notes |
+| ---                                                                                   | ---        | --- |
+| *Approve RFC*                                                                         |            |  |
+| ↳ [Yank crates with a reason](https://rust-lang.github.io/rust-project-goals/2024h2/yank-crates-with-a-reason.html#ownership-and-team-asks)   | [二手掉包工程师][] |  |
+| *Standard reviews*                                                                    |            |  |
+| ↳ [Yank crates with a reason](https://rust-lang.github.io/rust-project-goals/2024h2/yank-crates-with-a-reason.html#ownership-and-team-asks)   | [二手掉包工程师][] |  |
+| *Try it out in crates.io*                                                             |            |  |
+| ↳ [Yank crates with a reason](https://rust-lang.github.io/rust-project-goals/2024h2/yank-crates-with-a-reason.html#ownership-and-team-asks)   | [二手掉包工程师][] |  |
+
+### docs-rs team
+| Goal                                                                  | Owner      | Notes |
+| ---                                                                   | ---        | --- |
+| *Collecting popular queries for review*                               |            |  |
+| ↳ [Feedback and testing](https://rust-lang.github.io/rust-project-goals/2024h2/rustdoc-search.html#ownership-and-team-asks)   | [Michael Howell][] |  |
+
+### lang team
+| Goal                                                                                                                                  | Owner            | Notes              |
+| ---                                                                                                                                   | ---              | ---                |
+| *Design meeting*                                                                                                                      |                  |                    |
+| ↳ [Patterns of empty types](https://rust-lang.github.io/rust-project-goals/2024h2/Patterns-of-empty-types.html#ownership-and-team-asks)                                                       | [@Nadrieril][]       |                    |
+| ↳ ["Stabilizable" prototype for expanded const generics](https://rust-lang.github.io/rust-project-goals/2024h2/min_generic_const_arguments.html#ownership-and-team-asks)                      | [Boxy][]         | Up to 1, if needed |
+| ↳ [Experiment with relaxing the Orphan Rule](https://rust-lang.github.io/rust-project-goals/2024h2/Relaxing-the-Orphan-Rule.html#ownership-and-team-asks)                                     | ![Help wanted][] | Up to 1, if needed |
```

Comment:
I have some questions about the orphan rule goal, and I'm not sure where to put them. It strikes me as quite open ended and not obviously focused on the right constraints. (cc @joshtriplett as mentor)

For example, we could choose to relax the orphan rule only within a restricted set of co-versioned crates that we treat as "one big crate" for coherence purposes. This would not meet the axioms listed in the goal, but I _believe_ it would still improve things for a significant set of users.

If we instead go with visibility restrictions on impls, that might work and solve a larger subset, but I think the design will have to be guided by someone close to the implementation to be viable.

I would love to have a design meeting if a viable looking design emerges, but I want to make sure this feedback is taken into account before someone spends a lot of time on it.

Reply:
Due to @lcnr's concern, I plan to remove this goal from the next round of revisions (coming up soon). I'll add your points to the FAQ though to be incorporated later.